bbb/services.py

```diff
@@ -577,34 +578,38 @@ def handlePending(self, data, msg):
             self.bbb_db.deleteTask(taskid)
             return
 
-        # Lock the tasks table to prevent double scheduling
-        with lock_table(self.bbb_db.db, self.bbb_db.tasks_table.name):
-            our_task = self.bbb_db.getTask(taskid)
-
-            # When Buildbot Builds end up in a RETRY state they are automatically
-            # retried against the same BuildRequest. The BuildbotListener reflects
-            # this into Taskcluster be calling rerunTask, which creates a new Run
-            # for the same Task. In these cases, we don't want to do anything
-            # except update our own runId. If we created a new BuildRequest for it
-            # we'd end up with an extra Build.
-            if our_task:
-                # Taskcluster guarantees *at least* one message per event. Check
-                # runId to ignore duplicates
-                if our_task.runId >= runid:
-                    log.info("task %s run %s brid %s: ignoring duplicated message",
-                             taskid, runid, our_task.buildrequestId)
-                else:
-                    log.info("task %s: run %s: buildrequest %s: updating run id", taskid, runid, our_task.buildrequestId)
-                    self.bbb_db.updateRunId(our_task.buildrequestId, runid)
-            # If the task doesn't exist we need to insert it into our database.
-            # We don't want to claim it yet though, because that will mark the task
-            # as running. The BuildbotListener will take care of that when a slave
-            # actually picks up the job.
+        our_task = self.bbb_db.getTask(taskid)
+
+        # When Buildbot Builds end up in a RETRY state they are automatically
+        # retried against the same BuildRequest. The BuildbotListener reflects
+        # this into Taskcluster be calling rerunTask, which creates a new Run
+        # for the same Task. In these cases, we don't want to do anything
+        # except update our own runId. If we created a new BuildRequest for it
+        # we'd end up with an extra Build.
+        if our_task:
+            # Taskcluster guarantees *at least* one message per event. Check
+            # runId to ignore duplicates
+            if our_task.runId >= runid:
+                log.info("task %s run %s brid %s: ignoring duplicated message",
+                         taskid, runid, our_task.buildrequestId)
             else:
-                log.info("task %s: run %s: injecting task into bb", taskid, runid)
+                log.info("task %s: run %s: buildrequest %s: updating run id", taskid, runid, our_task.buildrequestId)
+                self.bbb_db.updateRunId(our_task.buildrequestId, runid)
```

Comment:
it handles buildbot RETRYs:  bblistener sees it, schedules another run, but we don't create a buildbot job in this case, because it's already running.